sunkit_image/enhance.py

```diff
@@ -0,0 +1,177 @@
+"""
+This module contains functions that can be used to enhance the entire solar image.
+"""
+
+import numpy as np
+import scipy.ndimage as ndimage
+
+__all__ = [
+    "mgn1",
+    "mgn2"
+]
+
+
+def mgn1(
+    data,
+    sigma=[1.25, 2.5, 5, 10, 20, 40],
+    k=0.7,
+    gamma=3.2,
+    h=0.7,
+    weights=None,
+    truncate=3
+):
+
+    """
+    Multi-scale Gaussian normalization.
+
+    Parameters
+    ----------
+    data : `numpy.ndarray`
```

Comment:
Wouldn't it be better that both input and output are ```sunpy.map``` rather than ```numpy.ndarray```

Reply:
I'm not sure we need to limit it to just a SunPy map. If the code works on a standard array, I think we should keep it that way.

Reply:
Okay

Reply:
I agree with nabobalis here.  The base algorithms should accept numpy arrays in general.  Later we can develop higher level functions that accept SunPy maps as arguments.

Reply:
Yes I am keeping them as numpy arrays

Reply:
We do want to support sunpy maps as well.

Reply:
use astropy units where appropriate.  For example, the sigma argument here should be `sigma=[1.25, 2.5, 5, 10, 20, 40]*u.pix`